iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```diff
@@ -33,6 +35,39 @@ SmallVector<Attribute, 4> indexSequence(int64_t n, MLIRContext *ctx) {
       }));
 }
 
+LogicalResult releaseRefIfLastUse(Operation *operation, Value listOperand) {
+  auto ctx = operation->getContext();
+  auto loc = operation->getLoc();
+
+  OpBuilder builder(operation);
+  builder.setInsertionPointAfter(operation);
+
+  auto funcOp = operation->getParentOfType<IREE::VM::FuncOp>();
+
+  // TODO(simon-camp): Cache the liveness analysis.
+  ValueLiveness liveness;
+  if (failed(liveness.recalculate(funcOp))) {
```

Comment:
TODO is fine - note that ValueLiveness is pretty much the slowest possible piece of code ever so FYI you'll hit slowdowns on anything bigger than a unit test.

There's also `mlir/Analysis/Liveness.h` upstream now which does most of the same stuff (I've been meaning to switch to it) - you may be able to use that here and at least not have to worry about my junk implementation :)